Plugins/BenchmarkTool/BenchmarkTool+CreateBenchmark.swift

```diff
@@ -0,0 +1,158 @@
+        var subPath = FilePath() // subpath rooted in package used for directory creation
+        subPath.append("Package.swift") // package/Benchmarks/targetName
+        outputPath.append(subPath.components)
+
+        // Write out benchmark boilerplate
+        do {
+            let fd = try FileDescriptor.open(
+                outputPath, .writeOnly, options: [.append], permissions: .ownerReadWrite
+            )
+
+            do {
+                try fd.closeAfter {
+                    do {
+                        try boilerplate.withUTF8 {
+                            _ = try fd.write(UnsafeRawBufferPointer($0))
+                        }
+                    } catch {
+                        print("Failed to write to file \(outputPath)")
+                    }
+                }
+            } catch {
+                print("Failed to close fd for \(outputPath) after write.")
+            }
+
+        } catch {
+            if errno == EPERM {
+                print("Lacking permissions to write to \(outputPath)")
+                print("Give benchmark plugin permissions by running with e.g.:")
+                print("")
+                print("swift package --allow-writing-to-package-directory benchmark init \(targetName)")
+                print("")
+            } else {
+                print("Failed to open file \(outputPath), errno = [\(errno)]")
+            }
+        }
+    }
+
+    mutating func createBenchmarkTarget() {
+        guard let targetName else {
+            failBenchmark("targetName not specified for createBenchmarkTarget()")
+            return
+        }
+
+        print("Creating benchmark \(targetName) in \(self.baselineStoragePath)")
+
+        var boilerplate = """
+        // Benchmark boilerplate generated by Benchmark
+
+        import Benchmark
+        import Foundation
+
+        let benchmarks = {
+            Benchmark("SomeBenchmark") { benchmark in
+                for _ in benchmark.scaledIterations {
+                    blackHole(Date()) // replace this line with your own benchmark
+                }
+            }
+            // Add additional benchmarks here
+        }
+
+        """
+
+        var outputPath = FilePath(baselineStoragePath) // package
+        var subPath = FilePath() // subpath rooted in package used for directory creation
+
+        subPath.append(benchmarksDirectory) // package/Benchmarks
+        subPath.append("\(targetName)") // package/Benchmarks/targetName
+
+        outputPath.createSubPath(subPath) // Create destination subpath if needed
+        outputPath.append(subPath.components)
+        outputPath.append("\(targetName).swift")
+
+        // Write out benchmark boilerplate, open with .exclusiveCreate to abort if file exists
+        do {
+            let fd = try FileDescriptor.open(
+                outputPath, .writeOnly, options: [.truncate, .create, .exclusiveCreate], permissions: .ownerReadWrite
```

Comment:
Need `.truncate` if we expect to create a new file?

Reply:
`.truncate` is there?

Reply:
` [.truncate, .create, .exclusiveCreate]`

Reply:
yep, but is it needed? [](edited-with-Axolo)

Reply:
Ah, no. Removed it.